packages/admin/src/pages/super-admin-dashboard/manage-departments/add/index.page.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  FlexibleQuestionPageLayout,
+  QuestionPageGetServerSideProps,
+  TextInput,
+} from 'gap-web-ui';
+import { GetServerSidePropsContext } from 'next';
+import CustomLink from '../../../../components/custom-link/CustomLink';
+import Meta from '../../../../components/layout/Meta';
+import getConfig from 'next/config';
+import InferProps from '../../../../types/InferProps';
+import { getUserTokenFromCookies } from '../../../../utils/session';
+import { createDepartmentInformation } from '../../../../services/SuperAdminService';
+import { Department } from '../../types';
+
+export function getServerSideProps(context: GetServerSidePropsContext) {
+  const fetchPageData = async (jwt: string) => {
+    return { jwt };
+  };
+
+  const handleRequest = async (body: Omit<Department, 'id'>, jwt: string) =>
+    createDepartmentInformation(body, jwt);
+
+  return QuestionPageGetServerSideProps({
+    context,
+    fetchPageData,
+    handleRequest,
+    jwt: getUserTokenFromCookies(context.req),
+    onErrorMessage: 'Failed to create department, please try again later.',
+    onSuccessRedirectHref: `/super-admin-dashboard/manage-departments/`,
+  });
+}
+
+const AddDepartmentPage = ({
+  formAction,
+  csrfToken,
+  fieldErrors,
```

Comment:
In the event of fieldErrors, `QuestionPageGetServerSideProps` should also pass in a prop called `previousValues`. We should use this to add defaults to our text inputs here, otherwise whatever the user inputted would be wiped after validation errors